src/apps/external_executor/external_executor.cpp

```diff
@@ -80,9 +84,24 @@ namespace externalexecutor
       {
         return executor_ids.size();
       }
+
+      void erase(ExecutorId to_remove)
+      {
+        auto it =
+          std::find(executor_ids.begin(), executor_ids.end(), to_remove);
+        while (it != executor_ids.end())
+        {
+          it = executor_ids.erase(it);
+          it = std::find(it, executor_ids.end(), to_remove);
+        }
+      }
     };
 
-    std::unordered_map<std::string, ExecutorIdList> supported_uris;
+    // Temporary implementation: Store supported uris on Register, insert into
+    // dispatch container on Activate
+    std::unordered_map<ExecutorId, std::vector<std::string>> supported_uris;
+    std::unordered_map<std::string, ExecutorIdList>
+      supported_uris_for_active_executors;
```

Comment:
This is why I think we should collapse `Register` and `Activate` into a single RPC, so we don't have to store the supported URIs independently from the dispatchable URIs.

Reply:
That sounds good to me